src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -62,11 +63,21 @@ class RoomSchemaLocationWorkaround implements Workaround {
             roomSchemaMergeLocations = roomExtension.roomSchemaMergeLocations
         }
 
-        JavaCompileWorkaround javaCompileRoomTask = JavaCompileWorkaround.create(project, roomExtension, mergeTask)
 
-        project.plugins.withId("kotlin-kapt") {
-            KaptWorkaround.create(project, roomExtension, mergeTask)
-            javaCompileRoomTask.javaCompileSchemaGenerationEnabled = false
+        // We favor ksp over kapt and apt in projects with multiple processors configured. If Ksp is applied and using
+        // the room-compiler, kapt and apt workarounds won't be applied.
+        if (project.configurations.find { it.name == "ksp" } != null) {
```

Comment:
why do we check the configurations instead of `project.plugins.withId("ksp") {` ?

would this code work if someone only added ksp dependencies to `kspDebug`?

Reply:
done https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/1d141065d6d95a97eb918209225e7cfd47047559